### PR TITLE
Able to click outside the Publish Commit dialog #6185

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6185

### CODE REVIEW

File: ui/app/src/components/GitManagement/RepoGrid/RepoGrid.tsx [hunk 1]
```diff
@@ -256,10 +263,11 @@ export function RepoGrid(props: RepoGridProps) {
...
         commitId={postPullState.mergeCommitId}
-        open={postPullState.openPublishCommitDialog}
-        onClose={() => {
-          setPostPullState({ openPublishCommitDialog: false });
-        }}
+        open={publishCommitDialogState.open}
+        hasPendingChanges={publishCommitDialogState.hasPendingChanges}
+        onWithPendingChangesCloseRequest={publishCommitDialogPendingChangesCloseRequest}
+        onSubmittingAndOrPendingChange={publishCommitDialogState.onSubmittingAndOrPendingChange}
+        onClose={publishCommitDialogState.onClose}
```
Comment (rart) [inline ui/app/src/components/GitManagement/RepoGrid/RepoGrid.tsx:265]: Shouldn't this rather be `publishCommitDialogPendingChangesCloseRequest`?

Or the `useWithPendingChangesCloseRequest()` call be inside the dialog and simply send onClose here?

Looks odd that it receives the two props. Let's check on various other dialogs and see if they are **all** this way.

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx [hunk 2]
```diff
@@ -145,7 +150,10 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
             <PublishOnDemandForm
               mode="git"
               formData={data}
-              setFormData={(newData) => setState(newData)}
+              setFormData={(newData) => {
+                fnRefs.current.onSubmittingAndOrPendingChange({ hasPendingChanges: true });
```
Comment (rart) [inline ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx]: I don't think this covers blanking the fields? (e.g. backspace)
Also, as soon as the dialog is opened, it has the commit id — the whole reason of this ticket — which should disallow closing. 

Make sure you're testing in Studio, not just on the isolated dev server app.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx]: I had a misunderstanding regarding the whole pre-filled id thing, I updated so now it won't close with the pre-filled commit id.

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx [hunk 3]
```diff
@@ -145,7 +155,15 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
             <PublishOnDemandForm
               mode="git"
               formData={data}
-              setFormData={(newData) => setState(newData)}
+              setFormData={(newData) => {
+                setState(newData);
+                const mergedData = { ...data, ...newData };
```
Comment (rart) [inline ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx:160]: What's this for? Doesn't newData come with comment, commidIds and pub target?

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx [hunk 4]
```diff
@@ -145,7 +155,15 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
...
+                if (isBlank(mergedData.comment) && isBlank(mergedData.commitIds)) {
+                  fnRefs.current.onSubmittingAndOrPendingChange({ hasPendingChanges: false });
+                } else {
+                  fnRefs.current.onSubmittingAndOrPendingChange({ hasPendingChanges: true });
+                }
```
Comment (rart) [inline ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx]:
```suggestion

                  fnRefs.current.onSubmittingAndOrPendingChange({ hasPendingChanges: !isBlank(mergedData.comment) || !isBlank(mergedData.commitIds) });
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx:160]: newData is an object with only the edited field, for example `{ comment: 'a comment' }`

**Review by rart — APPROVED** [review]

**Pull request merged**